Update MailPoet and Klaviyo variant assignment to 50/50 split

nerrad opened the pull request:
### Submission Review Guidelines:

-   [x] I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   [x] I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

This PR updates the variant assignment configuration for MailPoet and Klaviyo plugins to create an equal 50/50 split in visibility. Previously, the configuration was set to a 70/30 split (MailPoet: 70%, Klaviyo: 30%).

Closes #522.

### Screenshots or screen recordings:

| Before | After |
| ------ | ----- |
| MailPoet visible for 70% of stores (variant 1-84) | MailPoet visible for 50% of stores (variant 1-60) |
| Klaviyo visible for 30% of stores (variant 85-120) | Klaviyo visible for 50% of stores (variant 61-120) |

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up a fresh WooCommerce installation
2. Verify the `woocommerce_remote_variant_assignment` option is set (it should be a random number between 1-120)
3. Test with different variant assignments:
   - For values 1-60: Verify MailPoet is visible
   - For values 61-120: Verify Klaviyo is visible
4. Verify that the visibility is mutually exclusive (stores should only see one of the plugins)

### Testing that has already taken place:

Environment details:
- Local development environment
- WordPress 6.4
- WooCommerce trunk
- Default theme
- No additional plugins installed

Testing performed:
- Verified variant assignments work as expected for both plugins
- Confirmed no existing tests were affected as there were no specific tests for the previous 70/30 split
- Tested visibility rules for both plugins across the full range of variant assignments

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Update MailPoet and Klaviyo variant assignment to 50/50 split. This change updates the visibility configuration for MailPoet and Klaviyo plugins to create an equal 50/50 split in visibility, replacing the previous 70/30 split configuration.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

nerrad commented:
Hi @chihsuan - besides asking for your review, I have a couple additional questions/things I'd like you to confirm:

- I [see here](https://github.com/woocommerce/woocommerce/blob/febfbc4b24b0d3c9ee73c2fe8cca468c66b158e1/plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php#L99-L112) that if the `disable-core-profiler-fallback` is true, the core profiler fallback will never be shown for the set of users falling under 60 as the value for the `woocommerce_remote_variant_assignment` option. Effectively that means that for the change here, neither Klaviyo nor MailPoet would show up in the free plugin options if the fallback is invoked. Is that correct? If so, I'm not sure I understand the purpose of this logic. Whey would we never want to show plugins matching a variant under 60?
- I understand that along with the changes here, we need to update WooCommerce.com to serve the correct variants when requested remotely. I'll do a follow up PR there.

chihsuan reviewed (approved):
I’ve reviewed the changes and tested the PR locally—everything works as expected. 🚢

chihsuan commented:
> Effectively that means that for the change here, neither Klaviyo nor MailPoet would show up in the free plugin options if the fallback is invoked. Is that correct?
 
@nerrad Yes, that's correct.

> If so, I'm not sure I understand the purpose of this logic. Whey would we never want to show plugins matching a variant under 60?

The purpose of this logic is to test the impact of removing the core profiler fallback recommendations. 

We aim to eliminate the fallback recommendations due to the inefficiency of maintaining updates (Issue: https://github.com/woocommerce/woocommerce/issues/56080), as it requires updating both the fallback and remote endpoints.

The fallback is only used when:

1. "Show Suggestions" option is disabled in Advanced Settings. The default value is true and users are unlikely to disable it before completing the onboarding.
2. The remote endpoint is not available. We assume this scenario is rare and if any issue arises, we can fix it immediately without WC release.

> I understand that along with the changes here, we need to update WooCommerce.com to serve the correct variants when requested remotely. I'll do a follow up PR there.

Yes, thank you! 🙌

nerrad commented:
<!-- linear:replyTo:2849249009 -->

Okay so this is being tested for 50% of users initially. But the goal is to eventually remove any fallbacks.

chihsuan commented:
<!-- linear:replyTo:2849249009 -->

Yes, that's correct.

nerrad committed:
Change MailPoet and Klaviyo variant assignment to 50/50 split

- Update MailPoet visibility to show for variant numbers 1-60 (50% of stores)\n- Update Klaviyo visibility to show for variant numbers 61-120 (50% of stores)\n- Update comments to reflect the new split

Commit:
Add changefile(s) from automation for the following project(s): woocommerce
